devtools/maven/src/main/java/io/quarkus/maven/Deployer.java

```diff
@@ -61,29 +61,29 @@ public Optional<ArtifactDependency> getExtensionArtifact(MavenProject project) {
     /**
      * Get the deployer by name or the first one found in the project.
      *
-     * @project the project to search for deployer extensions
+     * @param project the project to search for deployer extensions
      * @return the {@link Optional} builder matching the name, project.
      */
     public static Optional<Deployer> getDeployer(MavenProject project) {
         return DeploymentUtil.getEnabledDeployer()
-                .or(() -> getProjecDeployer(project).stream().findFirst()).map(Deployer::valueOf);
+                .or(() -> getProjectDeployer(project).stream().findFirst()).map(Deployer::valueOf);
     }
 
     /**
      * Get teh deployer extensions found in the project.
      *
-     * @param the project to search for extensions
-     * @return A set with the discovered extenions.
+     * @param project The project to search for extensions
+     * @return A set with the discovered extensions.
      */
-    public static Set<String> getProjecDeployer(MavenProject project) {
+    public static Set<String> getProjectDeployer(MavenProject project) {
         return project.getDependencies().stream()
                 .filter(d -> QUARKUS_GROUP_ID.equals(d.getGroupId()))
                 .map(d -> strip(d.getArtifactId()))
-                .filter(n -> Arrays.stream(Deployer.values()).anyMatch(e -> e.equals(n)))
+                .filter(n -> Arrays.stream(Deployer.values()).anyMatch(e -> false))
```

Comment:
Could you explain this particular change?

Reply:
Thanks so much @gsmet  for the review, I can say that Result of 'e.equals(n)' is always 'false'  for the reason that  ‘equals' between objects of inconvertible types 'Deployer' and 'String’, So I made this shortcut 

I think this class deserves to write a test for it’s own 

Please let me know if I missed something with that, I can rollback this change the time to have some tests for Deployer enum

Reply:
Hum. This is a bug, and not a pretty one. I recommend you keep it the way it is for now and we will address it in a follow-up PR as we need to discuss it.

@iocanel I think you wanted to either compare with `e.name()` or `e.extension`. I'm not completely sure about the fix as I'm a bit unsure of how you handle knative. I don't see an extension for it so I suppose you would have kubernetes instead and that knative can't really be identified this way?

If so, it might be better to drop `.map(d -> strip(d.getArtifactId()))` entirely and compare on the extension rather than the name.

I will let you have a look.

Reply:
@gsmet, @iocanel I made a change on the last commit for this pull request on the related instruction, please let me know if you think it can be ok as a first try, I am ready to change it again to fit the desired solution, and on that topic I am in need of your help 


`
.filter(n -> Arrays.stream(Deployer.values()).anyMatch(e -> e.extension.equals(n)))
`

Thanks